### DISCUSSION TITLE
Is Kolasu compatible with antlr-kotlin?

### DISCUSSION THREAD

**Discussion by hans-d** [Strumenta/kolasu#327] [Q&A] [answered]
Seems that the Kolasu parser is not compatible when using antlr-kotlin used as target (incompatible types used for Parser). 

```
Type argument is not within its bounds: should be subtype of 'Parser'
```

Or am I doing something wrong / needs there be an additional step?

**Comment by ftomassetti** [comment, marked as answer]
Hi @hans-d thank you for your interest in Kolasu and ANTLR Kotlin.
The current version (1.5.*) is not compatible with ANTLR Kotlin, but branch 1.6 it is. It has been under development for many months. We are just waiting for Kotlin 2.0 to be officially released to release it, but if you want to look into it I can release an RC.

In general we made an explicit decision in ANTLR Kotlin to use different packages and classes, so you cannot just replace ANTLR Java with ANTLR Kotlin without some adaptations. In Kolasu 1.6 we support both, and if ANTLR Kotlin is used we can make parser that are multi-platform.
